### ISSUE TITLE
Cannot read source directory on v1.36

### ISSUE THREAD

**Issue by felixmoldovan** [ncw/rclone#1310]
Below the log output from a sync job. On v1.36 (including latest beta v1.36.b20) I get several messages of failure to read from source directory and the process would stop after three such failed attempts. It would typically be a different directory for each of the three errors.

However, when performing the same job with v1.35, the source directory is correctly read and the files are being sync'ed across.
Same config file, same environment, same source and destination et all... (Ubuntu 16.04)

Details here:
##### // Using v1.36 ... sync from source=onedrive to dest=google_cloud_storage
```
2017/04/04 07:24:35 ERROR : Attempt 1/3 failed with 0 errors and: error reading source directory "Music/aKING/Dutch Courage": directory not found
```
##### // Using v1.35 ... sync from source=onedrive to dest=google_cloud_storage
```
2017/04/04 13:15:48 One drive root '': Reading "Music/aKING/Dutch Courage/"
2017/04/04 13:15:59 Music/aKING/Dutch Courage/AKING - Decomposing Lullaby.mp3: Copied (new)
2017/04/04 13:16:01 Music/aKING/Dutch Courage/AKING - Holy Train.mp3: Copied (new)
2017/04/04 13:16:03 Music/aKING/Dutch Courage/AKING - I Believe.mp3: Copied (new)
2017/04/04 13:16:06 Music/aKING/Dutch Courage/AKING - Lonely Hands.mp3: Copied (new)
2017/04/04 13:16:08 Music/aKING/Dutch Courage/AKING - My Last Words.mp3: Copied (new)
2017/04/04 13:16:10 Music/aKING/Dutch Courage/AKING - Once I Arrive.mp3: Copied (new)
2017/04/04 13:16:11 One drive root '': Finished reading "Music/aKING/Dutch Courage/"
2017/04/04 13:16:13 Music/aKING/Dutch Courage/AKING - Safe As Houses.mp3: Copied (new)
2017/04/04 13:16:14 Music/aKING/Dutch Courage/AKING - Shine Your Light.mp3: Copied (new)
2017/04/04 13:16:17 Music/aKING/Dutch Courage/AKING - The Dance.mp3: Copied (new)
2017/04/04 13:16:19 Music/aKING/Dutch Courage/AKING - Thirsty For Love.mp3: Copied (new)
```
##### // Exact command used
```bash
./rclone --config /root/.config/rclone/rclone.conf --ignore-size --verbose --stats 0 sync f_onedrive:/ gcloud:f_backup
```

Thanks for any help

**Comment by naeloob** [comment]
What if you try with the --old-sync flag?

**Comment by felixmoldovan** [comment]
`--old-sync-method` (in case this is what you recommended) seems to do better.
- Is the new sync still in development / testing?
- What are the differences between old and new sync?
- Is the old-sync there to stay? Is it recommended to use it permanently or only as a workaround?
Thanks

**Comment by naeloob** [comment]
Yes, that's what i was talking about.
Well, i think these questions are on the @ncw side, but i think he wants to drop Old-Sync when New-Sync be fully tested.

**Comment by traynier** [comment]
This may well be due to a recent change in onedrive, breaking it for folders with spaces in the name?  See #1296 - I'm looking forward to a new build including the fix as rclone is largely unusable with onedrive for me currently.

**Comment by felixmoldovan** [comment]
###### Seeing that #1296 has been acknowledged as a bug, this can be closed and will watch progress there.
This seems to be indeed the pattern. Below the output from a sample source structure with and without spaces. Equally hopeful for a fix of this issue...
```
root@backup:~# rclone -vv sync f_onedrive:/Test/ gcloud:f_test_spaces
2017/04/05 13:34:36 DEBUG : rclone: Version "v1.36" starting with parameters ["/usr/bin/rclone.1.36" "--config" "/root/.config/rclone/rclone.conf" "-vv" "sync" "f_onedrive:/Test/" "gcloud:f_test_spaces"]
2017/04/05 13:34:37 DEBUG : f_onedrive: Saved new token in config file
2017/04/05 13:34:38 INFO  : Storage bucket f_test_spaces: Modify window is 1s
2017/04/05 13:34:38 DEBUG : One drive root 'Test': Reading ""
2017/04/05 13:34:38 DEBUG : One drive root 'Test': Finished reading ""
2017/04/05 13:34:38 DEBUG : One drive root 'Test': Reading "TestFolderWithNoSpaces/"
2017/04/05 13:34:38 DEBUG : One drive root 'Test': Finished reading "TestFolderWithNoSpaces/"
2017/04/05 13:34:39 INFO  : Storage bucket f_test_spaces: Waiting for checks to finish
2017/04/05 13:34:39 INFO  : Storage bucket f_test_spaces: Waiting for transfers to finish
2017/04/05 13:34:39 INFO  : TestFolderWithNoSpaces/TestFileWithNoSpaces.txt: Copied (new)
2017/04/05 13:34:39 ERROR : Storage bucket f_test_spaces: not deleting files as there were IO errors
2017/04/05 13:34:39 ERROR : Attempt 1/3 failed with 0 errors and: error reading source directory "Test Folder With Spaces": directory not found
2017/04/05 13:34:39 DEBUG : One drive root 'Test': Reading ""
2017/04/05 13:34:40 DEBUG : One drive root 'Test': Finished reading ""
2017/04/05 13:34:40 DEBUG : One drive root 'Test': Reading "TestFolderWithNoSpaces/"
2017/04/05 13:34:40 DEBUG : One drive root 'Test': Finished reading "TestFolderWithNoSpaces/"
2017/04/05 13:34:40 DEBUG : TestFolderWithNoSpaces/TestFileWithNoSpaces.txt: Size and modification time the same (differ by 0s, within tolerance 1s)
2017/04/05 13:34:40 DEBUG : TestFolderWithNoSpaces/TestFileWithNoSpaces.txt: Unchanged skipping
2017/04/05 13:34:40 INFO  : Storage bucket f_test_spaces: Waiting for checks to finish
2017/04/05 13:34:40 INFO  : Storage bucket f_test_spaces: Waiting for transfers to finish
2017/04/05 13:34:40 ERROR : Storage bucket f_test_spaces: not deleting files as there were IO errors
2017/04/05 13:34:40 ERROR : Attempt 2/3 failed with 0 errors and: error reading source directory "Test Folder With Spaces": directory not found
2017/04/05 13:34:40 DEBUG : One drive root 'Test': Reading ""
2017/04/05 13:34:40 DEBUG : One drive root 'Test': Finished reading ""
2017/04/05 13:34:41 DEBUG : One drive root 'Test': Reading "TestFolderWithNoSpaces/"
2017/04/05 13:34:41 DEBUG : One drive root 'Test': Finished reading "TestFolderWithNoSpaces/"
2017/04/05 13:34:41 DEBUG : TestFolderWithNoSpaces/TestFileWithNoSpaces.txt: Size and modification time the same (differ by 0s, within tolerance 1s)
2017/04/05 13:34:41 DEBUG : TestFolderWithNoSpaces/TestFileWithNoSpaces.txt: Unchanged skipping
2017/04/05 13:34:41 INFO  : Storage bucket f_test_spaces: Waiting for checks to finish
2017/04/05 13:34:41 INFO  : Storage bucket f_test_spaces: Waiting for transfers to finish
2017/04/05 13:34:41 ERROR : Storage bucket f_test_spaces: not deleting files as there were IO errors
2017/04/05 13:34:41 ERROR : Attempt 3/3 failed with 0 errors and: error reading source directory "Test Folder With Spaces": directory not found
2017/04/05 13:34:41 Failed to sync: error reading source directory "Test Folder With Spaces": directory not found
```
Would be nice an answer from @ncw to the questions about the  `--old-sync-method` current relevance and phase-out plans.

**Comment by ncw** [comment]
Sorry, I have been away - back now!

You can find a beta to fix #1296 in that issue.

My plan is to phase out --old-sync-method eventually, but not until I'm sure we don't need it any more.

**Comment by ncw** [comment]
Now that #1296 is done I'm going to close this.
